### PR TITLE
MTL-2396 Cobra+Viper Re-Integration and Application Refactor

### PR DESCRIPTION
**Pull request by rustydb**
### Summary and Scope

- Fixes: MTL-2396

#### Issue Type

- Bugfix Pull Request
- Docs Pull Request
- RFE Pull Request

### Prerequisites

- [ ] I have included documentation in my PR (or it is not required)
- [ ] I tested this on internal system (if yes, please include results or a description of the test)
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)

### CODE REVIEW

**Comment by rustydb** [comment]
Tests and lint finally passed. Now I'm reworking my commit, breaking it up and adding more descriptive messages.

**Comment by rustydb** [comment]
I am looking at `makedocs`, since it is more broken at this point.
I am also running through tests, generating configs for various systems in hpc-shasta-system-config.

**Comment by rustydb** [comment]
Flipping this back to a draft, until some of the non-code files are cleaned up to account for the refactor (e.g. Hugo).

**Comment by rustydb** [comment]
I think this is ready now.
I don't quite follow the new Hugo failure, if @jacobsalmela or someone has any insight I'd be happy to fix it here. Hugo is failing in `main` with the same error.

**Review by jpdavis-prof — APPROVED** [review]

File: testdata/upgrade-bss/csm1.0-csm1.2/csm1.0_bss_bootparameters.json [hunk 1]
```diff
@@ -568,7 +568,7 @@
             ],
             "config": {
               "confpath": "/etc/chrony.d/cray.conf",
```
Comment (jacobsalmela) [inline testdata/upgrade-bss/csm1.0-csm1.2/csm1.0_bss_bootparameters.json:570]: 😭

File: pkg/csi/csi.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+			configName := f.Name
+			// If using camelCase in the config file, replace hyphens with a camelCased string.
+			// Since viper does case-insensitive comparisons, we don't need to bother fixing the case, and only need to remove the hyphens.
+			if replaceHyphenWithCamelCase {
+				configName = strings.ReplaceAll(
+					f.Name,
+					"-",
+					"",
+				)
+			}
```
Comment (jacobsalmela) [inline pkg/csi/csi.go:167]: Might be worth having a sanitizer function to future proof any other oddly-named ones.  I can unfortunately see that happening.

**Review by jacobsalmela — APPROVED** [review]

**Pull request merged**